### DISCUSSION TITLE
z2m api

### DISCUSSION THREAD

**Discussion by lamboid** [Koenkk/zigbee2mqtt#7912] [General]
Hi all. 

I am wanting to develop some discovery code for IOT modelling and believe we have the perfect example here in z2m.

My hope is to poll the z2m for all devices and pull back the basic make & model data initially. 

Is there an api for z2m itself? (ideally RestAPI but anything would be marvellous)

Later I will be looking at eventing but that is for another day :P

**Comment by tavalin** [comment]
z2m stands for zigbee2mqtt and therefore you can communicate with it using mqtt.  See [here](https://www.zigbee2mqtt.io/information/mqtt_topics_and_message_structure.html#zigbee2mqttbridgedevices)  for the mqtt topic that contains information about devices
